[✨ feat] recoil을 적용한 카카오 Oauth 로그인 인증 기능 구현

shinsewon opened the pull request:
-. recoil, AsyncStorage을 적용하여, 카카오 로그인 구현
-. recoil/auth.ts 파일 생성 및 전역 상태 구현
-. hooks/useKakaoUser.ts, hooks/useKakaoAuthActions.ts로 임시 로그인,로그아웃 로직 구현
-. 임시작성 props 삭제
   1. 기존 App.tsx에서 임시로 뿌려주던 auth state 삭제
   2. RootStack에서 관리하던 isLoading State 전역 상태관리로 변경
   3. 전역 user 정보로 screen 분기처리를 위해 Stack.Navigator -> initialRouteName render방식을 삼항 연산자 render 방식으로 대체
-. MathTab에 BottomTab 두번째Tab 아이콘 변경
-. 로그아웃 작업을 위한 임시 프로필 Screen 구현
-. 불필요 주석 제거

중간에 커밋을 여러번해서 리뷰보실때 불편하실까봐 커밋들끼리 리베이스를 몇번했더니 조금 뒤죽박죽되었네요.. 
이제 적응 거의 다되서 다음엔 더 보기 편하게 해드릴게요!

shinsewon committed:
Update index.tsx
[✨ feat] recoil을 적용한 카카오 Oauth 로그인 인증 기능 구현

-. recoil, AsyncStorage을 적용하여, 카카오 로그인 구현
-. recoil/auth.ts 파일 생성 및 전역 상태 구현
-. hooks/useKakaoUser.ts, hooks/useKakaoAuthActions.ts로 임시 로그인,로그아웃 로직 구현
-. 임시작성 props 삭제
   1. 기존 App.tsx에서 뿌려주던 auth state 삭제
   2. RootStack에서 관리하던 isLoading State 전역 상태관리로 변경
   3. 전역 user 정보로 인한 screen 분기처리를 위해 Stack.Navigator initialRouteName 방식을 삼항 연산자로 대체

-. MathTab에 BottomTab 두번째Tab 아이콘 변경
-. 로그아웃 작업을 위한 임시 프로필 Screen 구현

shinsewon committed:
로그인 기능 구현중

dohye1 commented on src/recoil/atoms/auth.ts:
```diff
@@ -0,0 +1,19 @@
+import {atom} from 'recoil';
+
+export interface KakaoAuthUser {
```
책에사 인터페이스랑 type alias중 한가지로 통일해서 쓰는게 좋다고하더라구요

제가 맨 처음에 type alias는 enum타입의 값도 선언할수있다는 점이좋아서 type alias를 사용하고있었는데,
세원님은 따로 interface를 쓰신이유가 있나요?

dohye1 replied:
그리고 이 타입을 다른곳에서도 사용하려고하신다면 이 파일에서 export하는것보다 auth.d.ts라는 파일을 새로만들어서 그곳에 타입정의하는게 나을것같아요!

그럼 다른곳에서 import없이 쉽게 접근이 가능하니깐용

shinsewon replied:
저도 한가지만 사용하는게 좋다했는데, 저는 맨처음 배울때 interface가 확장성이 좋다고해서 그렇게 쓰고있었어요 ㅎㅎ
근데 말씀하신대로 type alias가 enum 타입때문에 좋다는걸 이번 프로젝트하면서 느끼긴했습니다.
아직까지 interface로 확장하는걸 많이 해보진 않았는데, 혜원님이 이야기하신대로 'type.d.ts'로 공통적인 타입부분들은 빼고, 필요할때만 type만 따로 설정해서 적용하는것도 나쁘지 않은 생각입니다.

혜원님 회사에서도 그럼 다  type alias 사용하시는건가요~?

dohye1 replied:
저희회사에서는 룰을 정하진않았는데 다 type을 쓰시긴 하더라구요
그리고 type도 확장하는 문법이있어서 인터페이스랑 기능적으로는 똑같은것같아요!!

shinsewon replied:
아 그렇군요! 확장문법 알아봐야겠네요!  저도 타입으로 사용할게요 ㅎㅎ

dohye1 commented on src/recoil/atoms/auth.ts:
```diff
@@ -0,0 +1,19 @@
+  accessToken: string;
+}
+
+export const authState = atom<KakaoAuthUser | null>({
+  key: 'kakaoAuthState',
+  default: null,
+});
```
지금 인증이 kakao인증 + local인증이 있는데, 
나중에 헷갈릴것같으니 authState > kakao~~관련된 이름으로 변경하시는게 어떨까요?

shinsewon replied:
네 그렇게 수정할게요 ㅎㅎ

dohye1 commented on src/screens/Auth/LoginScreen.tsx:
```diff
@@ -11,9 +11,26 @@ import {BorderedInput, CustomButton} from '@components/common';
 import AntIcon from 'react-native-vector-icons/AntDesign';
 
+import {useKakaoAuthActions} from 'hooks';
+import {KakaoAuthUser} from 'recoil/atoms';
```
타입을 이렇게받아와서 사용하는것보다 d.ts에 선언해서 글로벌타입으로 쓰는게 좋을것같아요

shinsewon replied:
네 반영 해볼게요!

dohye1 commented on src/screens/RootStack.tsx:
```diff
@@ -1,32 +1,45 @@
+  const {getKakaoLoginInfo} = useKakaoAuthActions();
+
+  useEffect(() => {
+    getKakaoLoginInfo();
```
isLoadingState의 값이 true인지 false인지 상관없이 호출이되도 되는건가요?
if문으로 특정조건일때만 실행되도록 해야하는거아닐까요?

shinsewon replied:
맞네요 굳이 state가 바뀔때마다 실행될 이유는 없는데 조건으로 구분처리 해놓을게요 ㅎㅎ

dohye1 commented on src/screens/RootStack.tsx:
```diff
@@ -1,32 +1,45 @@
 
-function RootStack({auth}: RootStackProps) {
-  const navigation = useNavigation<RootStackNavigationProps>();
+  if (!isLodingState) {
+    return <Loading />;
+  }
```
isLodingState가 true일때 loading 컴포넌트를 보여주는게 맞지않나요?
지금 실제값을 말씀드리는게아니라 isLoading이라는 이름만 생각해봤을때를 말씀드리는겁니당~

shinsewon replied:
네이밍적으로나, 흐름적으로 그게 맞는거 같습니다. 오해의 소지가 없도록 수정해 놓을게요~

dohye1 commented on src/recoil/atoms/auth.ts:
```diff
@@ -0,0 +1,19 @@
+  default: null,
+});
+
+export const isLoading = atom<boolean>({
+  key: 'isLoading',
+  default: false,
+});
```
이거 맨처음에 들어왔을때 로딩페이지가 보여져야하는데,
이름이 isLoading이면 defualt가 true여야하지않나요?

shinsewon replied:
이부분도 수정해놓을게요!

dohye1 commented on src/screens/MainTab/SettingStack/MyProfileScreen.tsx:
```diff
@@ -0,0 +1,45 @@
+import React, {useEffect} from 'react';
+import {useNavigation} from '@react-navigation/native';
+import {View, StyleSheet, Text, Pressable} from 'react-native';
+import {useKakaoAuthActions} from 'hooks';
+function MyProfileScreen() {
```
여기 한줄 띄워주는게 좋을것같아요~

shinsewon replied:
네 수정했습니다~~

dohye1 reviewed (commented):
recoil 구현하시느라 수고많으셨습니다.
전반적으로 틀은 잘잡으신것같아요

shinsewon committed:
MyProfileScreen 간격 띄움

shinsewon committed:
isLoading global state default value changed

shinsewon committed:
KakaoAuthState 전역 타입 설정
authState ->kakaoAuthState 네이밍 수정완료
isLodingState default value 변경완료
RootStack isLoading state 변경시  useEffect 분기처리 작업

dohye1 commented on src/hooks/useKakaoAuthActions.ts:
```diff
@@ -1,10 +1,10 @@
 import AsyncStorage from '@react-native-async-storage/async-storage';
 import {useSetRecoilState} from 'recoil';
-import {KakaoAuthUser, authState, isLoading} from 'recoil/atoms';
+import {kakaoAauthState, isLoading} from 'recoil/atoms';
```
kakaoAauthState
a가 2번들어갔는데 이거 수정하시면 될것같아요!

shinsewon committed:
KakaoAuthState 오탈자 수정완료
